Zend/tests/match/013.phpt

```diff
@@ -0,0 +1,93 @@
+--TEST--
+Match expression string jump table
+--INI--
+opcache.enable=1
+opcache.enable_cli=1
+opcache.opt_debug_level=0x20000
+--SKIPIF--
+<?php if (!extension_loaded('Zend OPcache')) die('skip Zend OPcache extension not available'); ?>
+--FILE--
+<?php
+
+function test($char) {
+    return match ($char) {
+        'a' => 'a',
+        'b', 'c' => 'b, c',
+        'd' => 'd',
+        'e', 'f' => 'e, f',
+        'g' => 'g',
+        'h', 'i' => 'h, i',
+    };
+}
+
+foreach (range('a', 'i') as $char) {
+    var_dump(test($char));
+}
+
+--EXPECTF--
+$_main:
+     ; (lines=%d, args=0, vars=1, tmps=2)
+     ; (after optimizer)
+     ; %s
+0000 INIT_FCALL 2 112 string("range")
+0001 SEND_VAL string("a") 1
+0002 SEND_VAL string("i") 2
+0003 V2 = DO_ICALL
+0004 V1 = FE_RESET_R V2 0013
+0005 FE_FETCH_R V1 CV0($char) 0013
+0006 INIT_FCALL 1 96 string("var_dump")
+0007 INIT_FCALL 1 112 string("test")
+0008 SEND_VAR CV0($char) 1
+0009 V2 = DO_UCALL
+0010 SEND_VAR V2 1
+0011 DO_ICALL
+0012 JMP 0005
+0013 FE_FREE V1
+0014 RETURN int(1)
+LIVE RANGES:
+     1: 0005 - 0013 (loop)
+
+test:
+     ; (lines=%d, args=1, vars=1, tmps=1)
+     ; (after optimizer)
+     ; %s
+0000 CV0($char) = RECV 1
+0001 SWITCH_STRING CV0($char) "a": 0020, "b": 0021, "c": 0021, "d": 0022, "e": 0023, "f": 0023, "g": 0024, "h": 0025, "i": 0025, default: 0026
+0002 T1 = IS_IDENTICAL CV0($char) string("a")
```

Comment:
For brand new specializations of `ZEND_MATCH_LONG` and `ZEND_MATCH_STRING`, I'd hope that the `IS_IDENTICAL` and `JMPNZ` checks could be removed completely, because the handler wouldn't need to go to the next opcode.

The `IS_IDENTICAL` and `JMPNZ` opcodes would still be needed for `ZEND_MATCH` to handle code like `match ($x) { null => 'something', [1, 2] => 'other', $otherVar => 'another thing' }`, though.

Reply:
> I'd hope that the IS_IDENTICAL and JMPNZ checks could be removed completely, because the handler wouldn't need to go to the next opcode.

If the optimization I mentioned for ZEND_MATCH_STRING worked, I'd expect this to happen